src/main/java/edu/kit/ipd/crowdcontrol/objectservice/moneytransfer/MoneyTransferManager.java

```diff
@@ -0,0 +1,262 @@
+
+        for (WorkerRecord worker : workers) {
+            List<GiftCodeRecord> payedCodesForWorker = chooseGiftCodes(worker, giftCodes);
+
+            giftCodes = workerBalanceOperations.getUnusedGiftCodes();
+            payWorker(worker, payedCodesForWorker);
+        }
+        if (giftCodes.size() < minGiftCodesCount) {
+            notificationText.append("There are less than ").append(minGiftCodesCount).append(" giftcodes in the database. It is recommended to add more.").append(System.getProperty("line.separator"));
+        }
+        LOGGER.trace("Completed submission of giftcodes to workers.");
+
+        //sends a notification about problems with submission of giftcodes
+        sendNotification();
+    }
+
+    private void fetchNewGiftCodes() throws MoneyTransferException {
+        Message[] messages;
+
+        //fetch new mails
+        try {
+            messages = mailHandler.fetchUnseen("inbox");
+        } catch (MessagingException e) {
+            throw new MoneyTransferException("The MailHandler couldn't fetch new giftcodes from the mailserver. " +
+                    "It seems, that there is either a problem with the server or with the properties file.");
+        }
+
+        //extract giftcodes and save them to the database
+        for (Message message : messages) {
+            try {
+                GiftCodeRecord rec = MailParser.parseAmazonGiftCode(message);
+                workerBalanceOperations.addGiftCode(rec.getCode(), rec.getAmount());
+            } catch (MoneyTransferException e) {
+                try {
+                    mailHandler.markAsUnseen(message);
+                    throw new MoneyTransferException(e.getMessage());
+                } catch (MessagingException f) {
+                    f.printStackTrace();
+                }
+            }
+        }
+    }
+
+    private List<GiftCodeRecord> chooseGiftCodes(WorkerRecord worker, List<GiftCodeRecord> giftCodes) {
+        List<GiftCodeRecord> payedCodes = new ArrayList<>();
+        int creditBalance = workerBalanceOperations.getBalance(worker.getIdWorker());
+
+        for (GiftCodeRecord nextCode : giftCodes) {
+            if (creditBalance == 0) {
+                break;
+            }
+            if (nextCode.getAmount() <= creditBalance) {
+                payedCodes.add(nextCode);
+                creditBalance -= nextCode.getAmount();
+            }
+        }
+
+        if (creditBalance >= payOffThreshold && creditBalance >= 15 && !payedCodes.isEmpty()) {
+            notificationText = notificationText.append("A worker has pending Payments in the amount of ").append(creditBalance).append("ct. Please add giftcodes, so the payment of the worker can be continued.").append(System.getProperty("line.separator"));
+        }
+        return payedCodes;
+    }
+
+    private void payWorker(WorkerRecord worker, List<GiftCodeRecord> giftCodes) throws MoneyTransferException {
+        if (!giftCodes.isEmpty()) {
```

Comment:
if gift codes are empty we are just doing nothing ? shouldnt we loadly complain "the world is on fire" ?